### PR TITLE
Docs fixes for v0.22.0 release

### PR DESCRIPTION
**Pull request by dime10**

### CODE REVIEW

File: pennylane/operation.py [hunk 1]
```diff
@@ -561,7 +561,7 @@ def compute_matrix(*params, **hyperparams):  # pylint:disable=unused-argument
         The canonical matrix is the textbook matrix representation that does not consider wires.
         Implicitly, this assumes that the wires of the operator correspond to the global wire order.
 
-        .. seealso:: :meth:`~.CNOT.matrix`
+        .. seealso:: :meth:`~.Operator.matrix`
```
Comment (albi3ro) [inline pennylane/operation.py:564]:
```suggestion
        .. seealso:: :meth:`~.Operator.get_matrix` and :func:`~.matrix`
```

Right now `matrix` is a property that will be deprecated, but there is a `qml.matrix` operator transform.

**Review by KetpuntoG — APPROVED** [review]
Seems good to me :)

File: pennylane/operation.py [hunk 2]
```diff
@@ -686,7 +686,7 @@ def compute_eigvals(*params, **hyperparams):
 
         Otherwise, no particular order for the eigenvalues is guaranteed.
 
-        .. seealso:: :meth:`~.RZ.eigvals`
+        .. seealso:: :meth:`~.Operator.eigvals`
```
Comment (albi3ro) [inline pennylane/operation.py:689]:
```suggestion
        .. seealso:: :meth:`~.Operator.get_eigvals` and :func:`~.eigvals`
```

same as above

File: pennylane/operation.py (hunk 1 above)
Comment (dime10) [reply pennylane/operation.py]: Oh right, that's why the link wasn't being generated 👍

File: pennylane/ops/snapshot.py [hunk 3]
```diff
@@ -33,6 +33,29 @@ class Snapshot(Operation):
...
+            qml.Snapshot()
+            return qml.expval(qml.PauliX(0))
+
+    >>> qml.snapshots(circuit)()
+    {0: array([1.+0.j, 0.+0.j, 0.+0.j, 0.+0.j]),
+    'very_important_state': array([0.70710678+0.j, 0.+0.j, 0.70710678+0.j, 0.+0.j]),
+    2: array([0.70710678+0.j, 0.+0.j, 0.+0.j, 0.70710678+0.j]),
+    'execution_results': array(0.)}
+
+    .. seealso:: :meth:`~.snapshots`
```
Comment (dime10) [inline pennylane/ops/snapshot.py:58]:
```suggestion
    .. seealso:: :func:`~.snapshots`
```

**Pull request merged**